mermaid_py/client.py

```diff
@@ -0,0 +1,443 @@
+import requests
+from .utilities import *
+
+
+class Client:
+    """
+    Client base class used for accessing MERMAID API.
+    """
+
+    # Class variables.
+    # Production MERMAID API root URL.
+    API_URL = "https://api.datamermaid.org/v1/"
+    # Development MERMAID API root URL.
+    API_DEV_URL = "https://dev-api.datamermaid.org/v1/"
+    # Non-project resource endpoints.
+    npr_endpoints = [
+        "health",
+        "managements",
+        "me",
+        "profiles",
+        "projecttags",
+        "sites",
+        "summarysites",
+        "version",
+    ]
+    # Attributes.
+    attrs_endpoints = [
+        "benthicattributes",
+        "fishfamilies",
+        "fishgenera",
+        "fishgroupings",
+        "fishsizes",
+        "fishspecies",
+    ]
+    # Project resources.
+    proj_resources = [
+        "collectrecords",
+        "managements",
+        "observers",
+        "project_profiles",
+        "sites",
+    ]
+    # Project observations.
+    proj_obs = [
+        "obsbenthiclits",
+        "obsbenthicpits",
+        "obscoloniesbleached",
+        "obshabitatcomplexities",
+        "obstransectbeltfishs",
+        "obsquadratbenthicpercent",
+    ]
+    # Project sample units, methods and events.
+    samples = [
+        "benthiclittransectmethods",
+        "benthicpittransectmethods",
+        "benthictransects",
+        "beltfishtransectmethods",
+        "bleachingquadratcollectionmethods",
+        "fishbelttransects",
+        "habitatcomplexitytransectmethods",
+        "quadratcollections",
+        "sampleevents",
+        "sampleunitmethods",
+    ]
+
+    def __init__(self, token=None, url=API_DEV_URL):
+        """
+        Constructor for Client base class used for accessing MERMAID API data.
+        :param token: (optional) Authenticated JWT token. Defaults to (token=None).
+        :type token: str
+        :param url: (optional) API URL. Defaults to (url='https://dev-api.datamermaid.org/v1/').
+        :type url: str
+        :return Client class object.
+        """
+        # Instance variables.
+        # MERMAID API URL.
+        self.url = url
+        # JWT token.
+        self.token = token
+        # Authenticated Client instance variable.
+        self.authenticated = False
+        # Checks if a token is provided and assigns authenticated instance variable.
+        if token:
+            self.authenticated = True
+        # Initializes requests Session and assigns headers for all Client MERMAID API calls.
+        self.session = requests.Session()
```

Comment:
See https://github.com/data-mermaid/mermaid-py/pull/1#discussion_r686232073 -- are you satisfied we're good with the fundamental data connection here?

Reply:
I believe this line creates the Session object not necessarily establishes the data connection, although inside _fetch_resource line 114 I added a 10-second timeout when sending the specifically prepared Request.

Reply:
Yeah I think there's nothing to do here after looking around a bit.